src/modules/tools/zprobe/ZProbe.cpp

```diff
@@ -507,3 +516,63 @@ void ZProbe::home()
     Gcode gc(THEKERNEL->is_grbl_mode() ? "G28.2" : "G28", &(StreamOutput::NullStream));
     THEKERNEL->call_event(ON_GCODE_RECEIVED, &gc);
 }
+
+// measure repeatability of probe (M48)
+void ZProbe::repeatability(StreamOutput *stream, int number_point, float x, float y) {
+  bool probe_result;
+  float mm;
+  float sum = 0.0f, mean = 0.0f, sigma = 0.0f, min = 99999.9f, max = -99999.9f, sample[number_point];
+  int nb_point_read = 0;
+
+  // some sanity checking
+  if (number_point<0) {
+    stream->printf("Wrong number point (P) !");
+    return;
+  }
+
+  // if x or y is passed, move to the position.
+  if (!isnan(x) || !isnan(y)) {
+    coordinated_move(x, y, NAN, this->fast_feedrate, false);
+  } else {
+    // first wait for all moves to finish
+    THEKERNEL->conveyor->wait_for_idle();
+  }
+
+  // probe the bed and store result until the number point is reached or an error is occured
+  stream->printf("Probe repeatability with %d sample, start sampling :\n", number_point);
+  for (int i = 0; i < number_point; i++) {
+
+      // if not setting Z then return probe to where it started, otherwise leave it where it is
+      probe_result = run_probe_return(mm, this->slow_feedrate, -1, false);
+
+      if(probe_result) {
+        // the result is in actuator coordinates moved
+        sample[i] = THEKERNEL->robot->from_millimeters(mm);
+        stream->printf(" sample %d, Z=%1.4f\n", i, sample[i]);
+        nb_point_read++;
+      } else {
+        stream->printf(" sample %d, can't probe !\n", i);
+        break;
```

Comment:
shouldn't this be return, as it would cause issues if not all samples were taken further on

Reply:
I would like keep the break instead of return :)

I want to loop until an error occured or until i have probed the good sample number.
During the loop, i increase the "nb_point_read". 
I compute statistical data with the "number of point read" after.

So i don't want exit the method, i just want to stop probe and compute the data with the number of sample done.

Reply:
Well then there will be a divide by zero in the case the first probe fails.